### ISSUE TITLE
Changes on current `mr create` and `issue create` flow

### ISSUE THREAD

**Issue by profclems** [profclems/glab#329]
I had a couple of thoughts about the current `mr create` and `issue create` flow and came up with these issues and suggested changes.

**Issues**
1. Description step is not skippable for interactive mode
1. Users are unable to preview the issue or merge request body as can be done on the UI

**Suggested Changes**

- [x] Make the description step skippable

```sh
$ glab mr create
? Title This is a test issue title
? Description (nano) [Enter to launch editor, s to skip] 
```

- [x] Provide submit options to allow users either submit the MR via the CLI or `continue in browser` or `cancel`

```sh
$ glab mr create
? Title This is a test title
? Body (nano) <Received>
? Submit?  [Use arrows to move, type to filter]
> Yes
  Continue in browser
  Cancel
```

This will open the browser to a prefilled merge request with the title and body the user submitted. Similar flow for `issue create`. 

Note on Prefilling issue with url query parameters: https://gitlab.com/gitlab-org/gitlab-foss/-/issues/19731#note_32449479

**Comment by zemzale** [comment]
Maybe we should also consider adding ability to write the body inline without opening the editor.
My workflow requires me to add `Resolve {JIRA-Ticket-ID}` to the body of the description. 
It's a bit annoying that I have to open an editor to just write one line. But we shouldn't remove the ability to open an editor because that is a nice thing to have if I need to write something more.

I just don't know what would be the best way to implement this in terms of UX.

**Comment by profclems** [comment]
You should be able to switch to writing the body inline by using the `--no-editor` flag. I guess this feature is less known
```sh
$ glab issue create --no-editor -t "This is my title"
? Description: [Enter 2 empty lines to finish] |
```
and yeah, you can completely avoid the interactive mode if you supply the required data using the option flags. If you use a flag for a particular required field, you will not be prompted for it. 
For instance `glab` will not prompt for title and description for the command:
```sh
$ glab issue create  -t "This is my title"  -d "Resolve {JIRA-Ticket-ID}"
```

**Comment by zemzale** [comment]
I really have missed the --no-editor flag, that fixes my issue

**Comment by profclems** [comment]
You can set an alias for that
```sh
glab alias set mrne 'mr create --no-editor'
```
and now you should be able to do, `glab mrne`

**Comment by profclems** [comment]
> 1. Make the description step skippable
> $ glab mr create
> ? Title This is a test issue title
> ? Description (nano) [Enter to launch editor, s to skip] 

This step has been implemented while working on #337
